autoemulate/emulators/gaussian_processs_utils/polynomial_features.py

```diff
@@ -0,0 +1,42 @@
+import itertools
+
+import numpy as np
+import torch
+
+
+class PolynomialFeatures:
```

Comment:
again, having a docstring and tests would be great here.

Reply:
Submitted the docstrings but to be honest I am not 100% how to go about making the tests.